mypy/binder.py

```diff
@@ -0,0 +1,288 @@
+from contextlib import contextmanager
+from typing import (Any, Dict, List, cast, Iterator)
+
+from mypy.types import Type, AnyType, PartialType
+from mypy.nodes import (Node, Var)
+
+from mypy.subtypes import is_subtype
+from mypy.join import join_simple, join_types
+from mypy.sametypes import is_same_type
+
+
+class Frame(Dict[Any, Type]):
+    """Frame for the ConditionalTypeBinder.
+
+    Frames actually in the binder must have
+    binder.frames[i].parent == binder.frames[i-1].
+
+    options_on_return represents a list of possible frames that we
+    will incorporate next time we are the top frame (e.g. frames from
+    the end of each if/elif/else block).
+
+    After a frame is complete, 'changed' represents whether anything
+    changed during the procedure, so a loop should be re-run.
+    'breaking' represents whether frame was necessarily breaking out.
```

Comment:
Agreed, though I have a subsequent clean-up in mind that should render this moot; at a minimum, it will give this variable a clearer name and purpose.